compiler/src/dotty/tools/dotc/core/Types.scala

```diff
@@ -1180,6 +1180,8 @@ object Types {
     def widenSingletons(using Context): Type = dealias match {
       case tp: SingletonType =>
         tp.widen
+      case tp: (TypeRef | AppliedType) if tp.typeSymbol.isAllOf(EnumCase) =>
```

Comment:
This doesn't widen only singleton cases, so it shouldn't be in a method named `widenSingletons`, it should probably be in its own method since you may want to widen singleton types but not enums.

Reply:
I've separated this out so it has its own widen step in ConstraintHandling